hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/server/http/BaseHttpServer.java

```diff
@@ -178,11 +180,14 @@ public BaseHttpServer(MutableConfigurationSource conf, String name)
       }
 
       String baseDir = conf.get(OzoneConfigKeys.OZONE_HTTP_BASEDIR);
-      if (!StringUtils.isEmpty(baseDir)) {
-        createDir(baseDir);
-        httpServer.getWebAppContext().setAttribute(JETTY_BASETMPDIR, baseDir);
-        LOG.info("HTTP server of {} uses base directory {}", name, baseDir);
+
+      if (StringUtils.isEmpty(baseDir)) {
+        baseDir =
+            conf.get(String.valueOf(getOzoneMetaDirPath(conf))) + serverDir;
```

Comment:
getOzoneMetaDirPath(conf) already returns the metadata path.  This path plus serverDir is the complete path. 

Please add a unit test in TestBaseHttpServer to verify this.

Reply:
Thanks @ChenSammi I have made the changes and added the UT in the `TestOzoneManagerHttpServer` that overrides the `BaseHttpServer`